Sources/Datadog/Core/Utils/DDError.swift

```diff
@@ -42,3 +42,13 @@ private func isNSErrorOrItsSubclass(_ error: Error) -> Bool {
     }
     return false
 }
+
+internal extension HTTPURLResponse {
+    func asError() -> Error? {
+        guard statusCode >= 400 && statusCode < 500 else {
+            return nil
+        }
+        let message = "\(statusCode) " + HTTPURLResponse.localizedString(forStatusCode: statusCode)
+        return NSError(domain: "HTTPURLResponse", code: statusCode, userInfo: [NSLocalizedDescriptionKey: message])
```

Comment:
I'm not sure on the use of `localizedString` and `NSLocalizedDescriptionKey` here. Using those for the error message makes it impossible to aggregate data across entire user base, allowing only for per-locale aggregation. We should use non-localized descriptions if possible.

Reply:
`NSError` has pretty much everything localized and we use it in `DDError` too. i think it's consistent to use `localizedString` here. aggregation shouldn't/can't be done with `userInfo/message`, they should use other values like `title/domain/code`

Reply:
That's a good point @ncreated, it's definitely going to appear in different languages but I assumed it was an ok expectations from our users (I've seen this many times with Crashlytics). In general it's likely that some instances of a an error will appear in `en_*`.
@buranmert also raises a good point, `DDError` has this localization problem already here while it potentially provides helpful information in the current locale.

In our case, the description is kind of a nice-to-have and not really required. HTTP status codes are well documented and it's not hard to get the description manually.

I think that it's reasonable to not include that part. The question remains then what is relevant for the message?

Reply:
Maybe this is sth we should have an overall look into in `RUMM-1042`. I left a note in this ticket to look holistically into errors analytics given the localization aspect.